provider/openstack/credentials.go

```diff
@@ -19,12 +19,15 @@ import (
 )
 
 const (
-	CredAttrTenantName = "tenant-name"
-	CredAttrUserName   = "username"
-	CredAttrPassword   = "password"
-	CredAttrDomainName = "domain-name"
-	CredAttrAccessKey  = "access-key"
-	CredAttrSecretKey  = "secret-key"
+	CredAttrTenantName        = "tenant-name"
+	CredAttrUserName          = "username"
+	CredAttrPassword          = "password"
+	CredAttrDomainName        = "domain-name"
+	CredAttrProjectDomainName = "project-domain-name"
+	CredAttrUserDomainName    = "user-domain-name"
+	CredAttrDefaultDomainName = "default-domain-name"
```

Comment:
IIANM, OS_DEFAULT_DOMAIN_NAME is just a shortcut for setting both the user and project domain names at once. Can we drop this from the attributes, but support ingesting $OS_DEFAULT_DOMAIN_NAME and setting both the attributes from it in DetectCredentials? I would prefer not to have multiple ways of specifying the attribute in the canonical representation.